Test/AppVeyorWaitForBuildJobs.Tests.ps1

```diff
@@ -0,0 +1,474 @@
+
+#Requires -Version 5.1
+Set-StrictMode -Version 'Latest'
+
+& (Join-Path -Path $PSScriptRoot -ChildPath 'Initialize-WhiskeyTest.ps1' -Resolve)
+
+$failed = $false
+$secret = $null
+$secretID = $null
+$whiskeyYml = $null
+[object[]]$jobs = $null
+$nextID = 10000000
+$currentJobID = $null
+$runByAppVeyor = $null
+$latestBuildID = $null
+
+function Get-NextID
+{
+    $ID = $nextID
+    $script:nextID++
+    return $ID
+}
+
+function GivenJob
+{
+    param(
+        [int]$WithID,
+
+        [Parameter(Mandatory,ParameterSetName='ForNotCurrentJob')]
+        [string]$WithStatus,
+
+        [Parameter(Mandatory,ParameterSetName='ForNotCurrentJob')]
+        [int]$ThatFinishesAtCheck,
+
+        [Parameter(Mandatory,ParameterSetName='CurrentJob')]
+        [Switch]$Current,
+
+        [Parameter(Mandatory,ParameterSetName='ForNotCurrentJob')]
+        [string]$WithFinalStatus,
+
+        [Parameter(ParameterSetName='ForNotCurrentJob')]
+        [switch]$ThatHasFinishedProperty
+    )
+
+    $script:jobs = & {
+        if( $jobs )
+        {
+            Write-Output $jobs
+        }
+
+        if( $WithID )
+        {
+            $jobID = $WithID
+        }
+        else
+        {
+            $jobID = Get-NextID
+        }
+
+        if( $Current )
+        {
+            $script:currentJobID = $jobID
+        }
+
+        $job = [pscustomobject]@{ 
+            'name' = ('job{0}' -f $jobID)
+            'jobId' = $jobID;
+            'status' = $WithStatus;
+            'checks' = $ThatFinishesAtCheck;
+            'finalStatus' = $WithFinalStatus;

```

Comment:
Nit: Unnecessary semicolons.

Reply:
Ack! I like my semicolons in hashtables. From my cold dead hands!